### PR TITLE
Make default query context show up in sql query logs

### PR DESCRIPTION
**Pull request by TSFenwick**
Fixes an issue where when you look at the sql query request logs emitted via an emitter or logged would be missing default query context values set in runtime.properties

#### Added DefaultQueryConfig to SqlLifecycle ...

This was chosen because while the default query context is added as part of QueryLifecycle it was never added as part of the sql lifecycle meaning it wouldn't show up in the logs or when emitted via an emitter. Since this is a duplicate value with the values being added at QueryLifecycle there should no worries about the eventual double write of default query context values

<hr>

##### Key changed/added classes in this PR
 * `SqlLifecycle`

<hr>

This PR has:
- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [x] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

File: sql/src/main/java/org/apache/druid/sql/SqlLifecycle.java [hunk 1]
```diff
@@ -120,7 +122,7 @@ public SqlLifecycle(
       RequestLogger requestLogger,
       QueryScheduler queryScheduler,
       AuthConfig authConfig,
-      long startMs,
+      DefaultQueryConfig defaultQueryConfig, long startMs,
```
Comment (TSFenwick) [inline sql/src/main/java/org/apache/druid/sql/SqlLifecycle.java:125]: new line

**Review by aggarwalakshay — APPROVED** [review]

**Review by kfaraz — COMMENTED** [review]
Thanks for the changes, @TSFenwick !

If I understand the problem correctly,
1. default query context params can be set as runtime properties
2. these are read by `QueryLifecycle` but not `SqlLifecycle`
3. SQL execution still honors these default params as it is converted to a native query at which point a `QueryLifecycle` gets involved
4. these default query params don't show up in logs or metrics emitted from the `SqlLifecycle`

Please correct me if I have missed anything. If this is the current behaviour, then your changes look good.

I would be nice if you could also add tests (or confirm that these tests already exist) to:
- ensure that SQL queries do indeed honor the default query params
- double writing of the default query params (once in `SqlLifecycle` and once in `QueryLifecycle`) does not affect behaviour

File: sql/src/main/java/org/apache/druid/sql/SqlLifecycleFactory.java [hunk 2]
```diff
@@ -35,21 +36,24 @@
...
 
   @Inject
   public SqlLifecycleFactory(
       PlannerFactory plannerFactory,
       ServiceEmitter emitter,
       RequestLogger requestLogger,
       QueryScheduler queryScheduler,
-      AuthConfig authConfig
+      AuthConfig authConfig,
+      DefaultQueryConfig defaultQueryConfig
```
[the remark this answers is not in the document]
Comment (TSFenwick) [reply sql/src/main/java/org/apache/druid/sql/SqlLifecycleFactory.java:48]: I'll get that done asap

**Comment by TSFenwick** [comment]
@kfaraz I'll see what I can do about implementing those two tests you want

File: sql/src/test/java/org/apache/druid/sql/avatica/DruidAvaticaHandlerTest.java [hunk 3]
```diff
@@ -237,6 +242,8 @@ public void configure(Binder binder)
     clientLosAngeles = DriverManager.getConnection(url, propertiesLosAngeles);
   }
 
+
```
Comment (kfaraz) [inline sql/src/test/java/org/apache/druid/sql/avatica/DruidAvaticaHandlerTest.java:245]: Nit: extra newlines.

File: sql/src/test/java/org/apache/druid/sql/avatica/DruidAvaticaHandlerTest.java [hunk 4]
```diff
@@ -214,6 +218,7 @@ public void configure(Binder binder)
                       .toProvider(QuerySchedulerProvider.class)
                       .in(LazySingleton.class);
                 binder.bind(QueryMakerFactory.class).to(NativeQueryMakerFactory.class);
+                binder.bind(new TypeLiteral<Supplier<DefaultQueryConfig>>(){}).toInstance(Suppliers.ofInstance(new DefaultQueryConfig(ImmutableMap.of())));
```
Comment (kfaraz) [inline sql/src/test/java/org/apache/druid/sql/avatica/DruidAvaticaHandlerTest.java:221]: Nit: style
Maybe break up this line so that it's easier to read.

**Review by kfaraz — APPROVED** [review]
Minor nitpicks. Overall LGTM.

**Pull request merged**